test/test_image_science.rb

```diff
@@ -207,4 +211,30 @@ def test_rotate_not_45
 
     assert_equal 'Angle must be 45 degree skew', e.message
   end
+
+  def test_buffer
+    buffer = nil
+    ImageScience.with_image @path do |img|
+      img.thumbnail(128) do |thumb|
+        img.save(@tmpjpeg)
+        buffer = img.buffer('.jpg')
+      end
+    end
+
+    file_data = File.new(@tmpjpeg).binmode.read
+    assert_equal file_data, buffer
+  end
+
+  def test_buffer_default
+    buffer = nil
+    ImageScience.with_image @path do |img|
+      img.thumbnail(128) do |thumb|
+        img.save(@tmppath)
+        buffer = img.buffer
+      end
+    end
+
+    file_data = File.new(@tmppath).binmode.read
```

Comment:
Oh. also, binmode probably doesn't work on 1.8... this won't be allowed until I officially drop 1.8

ETA: I can deal with that tho.